components/file_attachment_list/file_attachment_list.tsx

```diff
@@ -94,7 +71,7 @@ export default class FileAttachmentList extends React.PureComponent<Props, State
                         key={fileInfo.id}
                         fileInfo={sortedFileInfos[i]}
                         index={i}
-                        handleImageClick={this.handleImageClick}
+                        handleImageClick={(index) => this.handleImageClick(index, sortedFileInfos, postId)}
```

Comment:
Did you happen to figure out why the values from props were undefined in `handleImageClick` after your other changes?

Like I mentioned, it'd be better if we could avoid generating new functions and passing them as props into other redux components so that we can avoid all of the children of a component re-rendering when the parent re-renders

Reply:
@hmhealey can you try looking into converting to a normal function instead of arrow function, I did it a way of passing the paramaters as props to component and calling the function with those passed in props https://github.com/mattermost/mattermost-webapp/pull/9334/commits/5310cf9668d064a3ce2beb3265ee13dadbeb8916

Reply:
so it seems normal function was working too , thank you for https://github.com/mattermost/mattermost-webapp/pull/9334/commits/31e93809e071074d27eda3ac98042860727cb03e